ide/xmlprotocol.ml

```diff
@@ -117,16 +117,18 @@ let to_box = let open Pp in
       | x        -> raise (Marshal_error("*ppbox",PCData x))
     )
 
+let of_protected_string s = of_string (String.escaped_if_non_utf8 s)
+
 let rec of_pp (pp : Pp.t) = let open Pp in match Pp.repr pp with
     | Ppcmd_empty         -> constructor "ppdoc" "empty"  []
-    | Ppcmd_string s      -> constructor "ppdoc" "string" [of_string s]
+    | Ppcmd_string s      -> constructor "ppdoc" "string" [of_protected_string s]
     | Ppcmd_glue sl       -> constructor "ppdoc" "glue"   [of_list of_pp sl]
     | Ppcmd_box (bt,s)    -> constructor "ppdoc" "box"    [of_pair of_box of_pp (bt,s)]
-    | Ppcmd_tag (t,s)     -> constructor "ppdoc" "tag"    [of_pair of_string of_pp (t,s)]
+    | Ppcmd_tag (t,s)     -> constructor "ppdoc" "tag"    [of_pair of_protected_string of_pp (t,s)]
```

Comment:
I suppose this is excessive since any non-ASCII tag would most certainly be an error.